src/libmain/progress-bar.cc

```diff
@@ -365,6 +368,62 @@ class ProgressBar : public Logger
         writeToStderr("\r" + filterANSIEscapes(line, false, width) + ANSI_NORMAL + "\e[K");
     }
 
+		void draw_multi(State & state)
+		{
+			state.haveUpdate = false;
+			if (!state.active) return;
+
+			std::string line;
+			auto width = getWindowSize().second;
+			if (width <= 0) width = std::numeric_limits<decltype(width)>::max();
+
+			std::string status = getStatus(state);
+			if (!status.empty()) {
+				line += '[';
+				line += status;
+				line += "]";
+			}
+
+      #if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+			  system("cls");
+			#else
+        writeToStderr("\r\033[K\r");
+	      for (auto i = 1; i < state.lastLines; i++)
+	        writeToStderr("\033[1A\r\033[K\r");
+      #endif
+			writeToStderr(filterANSIEscapes(line, false, width));
+
+			state.lastLines = 1;
+
+			if (!state.activities.empty()) {
+				for (auto i = state.activities.begin(); i != state.activities.end(); ++i) {
+					if (!i->visible || (i->s.empty() && i->lastLine.empty())) {
+						continue;
+					}
+					line = i->s;
+
+				  if (!i->phase.empty()) {
+						line += " (";
+						line += i->phase;
+						line += ")";
+					}
+					if (!i->lastLine.empty()) {
+						if (!i->s.empty()) line += ": ";
+						line += i->lastLine;
+					}
+					state.lastLines += 1;
+					writeToStderr("\n" + filterANSIEscapes(line, false, width));
+				}
+			}
+		}
+
+		void draw(State & state)
+		{
+			if (settings.isExperimentalFeatureEnabled("progress-bar")) {
```

Comment:
What's the reason to hide this behind such a feature flag?

Reply:
This should be changed once #3961 lands